### PR TITLE
fix #50: cancel handled touch actions

### LINKED ISSUE #50: cancel handled touchEnd events
Currently it is not possible to integrate this library to an application that handles for example swipe events on an outer element.
Imagine you want to use this library inside an image slider with touch support and the "wipe to next image" event is also triggered if you want to move your zoomed in image.

So it is necessary to cancel the touchEnd event, if it is handled by the library and let it bubble if not.

Cancel if:
* user performed a zoom interaction
* user performed a drag interaction if zoom factor is greather than 1

### PR DESCRIPTION
**Pull request by martinkutter**
…ize changes

### CODE REVIEW

File: src/PinchZoom/component.tsx [hunk 1]
```diff
@@ -796,6 +796,11 @@ class PinchZoom extends React.Component<Props> {
   private _handlerOnTouchEnd = this._handlerIfEnable(
     (touchEndEvent: TouchEvent) => {
       this._fingers = touchEndEvent.touches.length;
+
+      if (isZoomInteraction(this._interaction) || (isDragInteraction(this._interaction) && this._zoomFactor !== 1)) {
```
Comment (davydspotnana) [inline src/PinchZoom/component.tsx:800]: I worry about backward compatibility, can we add a new prop to enable this logic?

```tsx
if(props.isXXXXX){
  if (isZoomInteraction(this._interaction) || (isDragInteraction(this._interaction) && this._zoomFactor !== 1)) {
    cancelEvent(touchEndEvent);
  }
}
```

**Comment by retyui** [comment]
Note: It is possible to drag an image with `zoom factor = 1` when `verticalPadding` & `horizontalPadding` aren't zero:

https://user-images.githubusercontent.com/95620376/165758316-4fbe4d10-7511-4363-85c9-20c634cffaf6.mp4


Could you please handle this case too?

File: src/PinchZoom/component.tsx [hunk 2]
```diff
@@ -796,6 +796,11 @@ class PinchZoom extends React.Component<Props> {
...
+        cancelEvent(touchEndEvent);
+      }
```
Comment (retyui) [inline src/PinchZoom/component.tsx:802]: I worry about backward compatibility, can we add a new prop to enable this logic?

```tsx
if(props.isXXXXX){
  if (isZoomInteraction(this._interaction) || (isDragInteraction(this._interaction) && this._zoomFactor !== 1)) {
    cancelEvent(touchEndEvent);
  }
}
```

**Comment by martinkutter** [comment]
Thank you for your fast feedback. Any recommendations for the prop name?
Would `shouldCancelHandledTouchEndEvents?: boolean` default `false` fit?

**Comment by martinkutter** [comment]
I changed the behavior to cancel only events when visual changes habe been made and added the props to enable it.

**Pull request merged**